.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: "build"
+
+env:
+  INFURA_API_KEY: ${{ secrets.INFURA_API_KEY }}
+  MNEMONIC: ${{ secrets.MNEMONIC }}
```

Comment:
These 2 environment variables are needed as repository secret to work.

As already discussed, otherwise we would need to modify the logic within `hardhat.config.ts`.